src/System/Components/PageLoadingBar.tsx

```diff
@@ -68,31 +72,35 @@ const easeOutExpo = "cubic-bezier(0.19, 1, 0.22, 1)"
 
 const loadingAnimation = keyframes`
   from {
-    width: 0%;
-    left: -15%;
+    transform: translateX(-10%) scaleX(0);
   }
   to {
-    width: 30%;
-    left: 0%;
+    transform: translateX(0) scaleX(0.3);
   }
 `
 
 const completeAnimation = keyframes`
   to {
-    width: 100%;
+    transform: scaleX(1);
   }
 `
 
 const fadeInAnimation = keyframes`
   from {
-    opacity: 0;
+    opacity: 1;
   }
   to {
     opacity: 1;
   }
 `
 
-const LoadingBar = styled(Box)<{ loading: string; isComplete: boolean; width }>`
+const LoadingBar = styled(Box)<{
+  loading: string
+  isComplete: boolean
+  scaleX: number
+}>`
+  transform-origin: left;
+  transform: scaleX(${({ scaleX }) => scaleX});
```

Comment:
I'd just set this direct with a `style` prop

Reply:
...this is necessary to take the out transition from where its eased to, regardless of how much has already loaded.

Reply:
Oh sure, I'm saying you can set that value using style so that it's not generating a ton of classes.

Reply:
But if it's just 2 values then it doesn't matter